sp_mock/backend/server.js

```diff
@@ -37,6 +37,10 @@ app.post("/", (req, res)=>{
     res.send("Server has registered a POST.")
 })
 
+app.post("/api/message", (req, res)=>{
```

Comment:
Good start but the service provider shouldn't receive the 'ack' in the backend. It is a message between the Interceptor and the Middleware only.

Notice that the the Service Provider never gets the ECDHInit messages. These messages terminate in the middleware.go